modules/adtrgtmeBidAdapter.js

```diff
@@ -1,332 +1,358 @@
+    ) {
       return true;
     } else {
-      logWarn('Adtrgtme bidder params missing or incorrect');
+      logWarn('Adtrgtme request invalid');
       return false;
     }
   },
 
-  buildRequests: function(validBidRequests, bidderRequest) {
-    if (isEmpty(validBidRequests) || isEmpty(bidderRequest)) {
+  buildRequests: function (bR, aR) {
+    if (isEmpty(bR) || isEmpty(aR)) {
       logWarn('Adtrgtme Adapter: buildRequests called with empty request');
       return undefined;
-    };
+    }
 
-    const requestOptions = {
+    const options = {
       contentType: 'application/json',
-      customHeaders: {
-        'x-openrtb-version': '2.5'
-      }
+      withCredentials: hasPurpose1Consent(
+        aR.gdprConsent
+      )
     };
 
-    requestOptions.withCredentials = hasPurpose1Consent(bidderRequest.gdprConsent);
-
     if (config.getConfig('adtrgtme.singleRequestMode') === true) {
-      const payload = generateOpenRtbObject(bidderRequest, validBidRequests[0]);
-      validBidRequests.forEach(bid => {
-        appendImpObject(bid, payload);
+      const data = createORTB(aR, bR[0]);
+      bR.forEach((bid) => {
+        appendImp(bid, data);
       });
 
-      return generateServerRequest({payload, requestOptions, bidderRequest});
+      return createRequest({ data, options, bidderRequest: aR });
     }
 
-    return validBidRequests.map(bid => {
-      const payloadClone = generateOpenRtbObject(bidderRequest, bid);
-      appendImpObject(bid, payloadClone);
+    return bR.map((b) => {
+      const data = createORTB(aR, b);
+      appendImp(b, data);
 
-      return generateServerRequest({payload: payloadClone, requestOptions, bidderRequest: bid});
+      return createRequest({
+        data,
+        options,
+        bidderRequest: b,
+      });
     });
   },
 
-  interpretResponse: function(serverResponse, { data, bidderRequest }) {
-    const response = [];
-    if (!serverResponse.body || !Array.isArray(serverResponse.body.seatbid)) {
-      return response;
+  interpretResponse: function (sR, { data, bidderRequest }) {
+    const res = [];
+    if (!sR.body || !Array.isArray(sR.body.seatbid)) {
+      return res;
     }
 
-    let seatbids = serverResponse.body.seatbid;
-    seatbids.forEach(seatbid => {
-      let bid;
-
+    sR.body.seatbid.forEach((sb) => {
       try {
-        bid = seatbid.bid[0];
+        let b = sb.bid[0];
+
+        res.push({
+          adId: deepAccess(b, 'adId') ? b.adId : b.impid || b.crid,
+          ad: b.adm,
+          adUnitCode: bidderRequest.adUnitCode,
+          requestId: b.impid,
+          cpm: b.price,
+          width: b.w,
+          height: b.h,
+          mediaType: BANNER,
+          creativeId: b.crid || 0,
+          currency: b.cur || DEFAULT_CUR,
+          dealId: b.dealid ? b.dealid : null,
+          netRevenue: true,
+          ttl: getTtl(bidderRequest),
+          meta: {
+            advertiserDomains: b.adomain || [],
```

Comment:
Networkid can be very helpful if you have it

Reply:
Thx, but we use site.id instead